### PR TITLE
Stability enhanced + added test case for authorisation code

### PR DESCRIPTION
**Pull request by danishjamal104**
## Description

* Added some logs for making debugging easier.
* New endpoint for testing SLCB. (**This endpoint is totally ment for internal testing and need not to be exposed for clients.**)
* Updated algorithm for encrypting the data using RSA encryption key.
* Added a new decrypt method in `ScurityUtils.java` class.
* Added cucumber test support.
* Added test for testing the authorisationCode using cucumber.
* Added cucumber test case for SLCB integration test.
* Added new properties file `application-test.yml` for test specific profile.

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Design related bullet points or design document link related to this PR added in the description above. 

- [x] Updated corresponding Postman Collection or Api documentation for the changes in this PR.

- [x] Create/update unit or integration tests for verifying the changes made.

- [x] Add required Swagger annotation and update API documentation with details of any API changes if applicable

- [x] Followed the naming conventions as given in https://docs.google.com/document/d/1Q4vaMSzrTxxh9TS0RILuNkSkYCxotuYk1Xe0CMIkkCU/edit?usp=sharing

### CODE REVIEW

**Comment by danishjamal104** [comment]
@fynmanoj The encryption keys and decryption keys used in the test case are only meant for this specific test and it doesn't expose the actual keys used in SIT or Prod environment.

File: build.gradle [hunk 1]
```diff
@@ -60,3 +70,29 @@ publishing {
...
+}
+
+
+configurations {
+    cucumberRuntime {
+        extendsFrom testImplementation
+    }
+}
+
+task cucumberCli() {
```
Comment (fynmanoj) [inline build.gradle:85]: please update README.md with details about how to use this task

File: src/main/java/org/mifos/connector/slcb/utils/SLCBUtils.java [hunk 2]
```diff
@@ -77,4 +78,18 @@ public static Payee convertTransactionToPayee(Transaction transaction) {
         return payee;
     }
 
+    public static int getStatusCodeOrDefault(Status status) {
+        if (status != null) {
+            return status.getCode();
+        }
+        return -1;
```
Comment (fynmanoj) [inline src/main/java/org/mifos/connector/slcb/utils/SLCBUtils.java:85]: use an invalid status enum for better readability,  something like `Status.INVALID.getCode()`

**Review by fynmanoj — CHANGES_REQUESTED** [review]

File: src/test/java/org/mifos/connector/slcb/cucumber/SLCBIntegrationStepDef.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+    private Exchange exchange;
+
+    private PaymentRequestDTO paymentRequestDTO;
+
+    @Autowired
+    ProducerTemplate template;
+
+    @Given("I have a batchId: {string}, requestId: {string}, purpose: {string}")
+    public void i_have_required_data(String batchId, String requestId, String purpose){
```
[the remark this answers is not in the document]
Comment (danishjamal104) [reply src/test/java/org/mifos/connector/slcb/cucumber/SLCBIntegrationStepDef.java:22]: ok Ill update this

File: build.gradle (hunk 1 above)
Comment (danishjamal104) [reply build.gradle:85]: noted

File: src/main/java/org/mifos/connector/slcb/utils/SLCBUtils.java (hunk 2 above)
Comment (danishjamal104) [reply src/main/java/org/mifos/connector/slcb/utils/SLCBUtils.java:85]: noted

File: src/test/java/org/mifos/connector/slcb/cucumber/SLCBIntegrationStepDef.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (danishjamal104) [reply src/test/java/org/mifos/connector/slcb/cucumber/SLCBIntegrationStepDef.java]: @fynmanoj done, please recheck

File: build.gradle (hunk 1 above)
Comment (danishjamal104) [reply build.gradle:85]: @fynmanoj done, please recheck

File: src/main/java/org/mifos/connector/slcb/utils/SLCBUtils.java (hunk 2 above)
Comment (danishjamal104) [reply src/main/java/org/mifos/connector/slcb/utils/SLCBUtils.java]: @fynmanoj done, please recheck

**Review by fynmanoj — APPROVED** [review]

**Pull request merged**